app/auth/components/Username.tsx

```diff
@@ -0,0 +1,32 @@
+type Props = {
+  setUserData: React.Dispatch<React.SetStateAction<UserData>>;
+  RegExp: boolean | null;
+};
+
+function Username({ setUserData, RegExp }: Props) {
+  return (
+    <div className="py-1 flex flex-col">
+      <label htmlFor="username">Username</label>
+      <input
+        type="text"
+        id="username"
+        className="auth_input"
+        style={{
+          borderColor: RegExp === false ? "red" : "",
+        }}
+        placeholder="User0046"
+        onChange={(e) =>
+          setUserData((prev) => ({ ...prev, username: e.target.value }))
+        }
+        required
+      />
+      {RegExp === false && (
```

Comment:
RegExp seems a bit confusing. Maybe we need to rename it to something like `isUserNameValid`?

Reply:
Sure, I will change it.